src/main/java/io/split/android/client/service/mysegments/MySegmentsSyncTask.java

```diff
@@ -121,6 +142,75 @@ public SplitTaskExecutionInfo execute() {
         return SplitTaskExecutionInfo.success(mTaskType);
     }
 
+    private void fetch(int initialRetries) throws HttpFetcherException, InterruptedException {
+        int remainingRetries = initialRetries;
+        mBackoffCounter.resetCounter();
+        while (remainingRetries > 0) {
+            AllSegmentsChange response = mMySegmentsFetcher.execute(getParams(false), getHeaders());
+            if (response == null) {
+                throw new HttpFetcherException("", "Response is null");
+            }
+
+            if (isStaleResponse(response)) {
+                long waitMillis = TimeUnit.SECONDS.toMillis(mBackoffCounter.getNextRetryTime());
+                Thread.sleep(waitMillis);
+                remainingRetries--;
+            } else {
+                updateStorage(response);
+                return;
+            }
+        }
+
+        AllSegmentsChange response = mMySegmentsFetcher.execute(getParams(true), getHeaders());
```

Comment:
Maybe you can create a function with this call to execute and the one in line 149.